### PR TITLE
Fail calling Cluster() on windows, skip tests appropriately

### PR DESCRIPTION
**Pull request by mattip**
## Why are these changes needed?

Multi-node mode (clusters) are broken on windows. This PR raises an error if Cluster() is called. It also xfails the appropriate tests.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by carlogrisetti** [comment]
What tests would you help sort this out? Last time I needed to spread training on multiple machines it worked fine.

(I do not want to mistake this for another functionality, like `ray up` does not work and I do not expect it to)

**Comment by mattip** [comment]
The test that started me down the path to reject clustering is `test_legacy_spillback_distribution` from `test_scheduling.py`

**Comment by mattip** [comment]
Replaces #20023.

File: python/ray/cluster_utils.py [hunk 1]
```diff
@@ -96,6 +97,8 @@ def __init__(self,
             shutdown_at_exit (bool): If True, registers an exit hook
                 for shutting down all started processes.
         """
+        if cluster_not_supported:
+            raise RuntimeError("cannot use Cluster on this platform")
```
Comment (pcmoritz) [inline python/ray/cluster_utils.py:101]: Ok, given what @carlogrisetti is saying, let's make this a logger.warning with a message like
```
Ray cluster mode is currently experimental and untested on Windows. If you are using it and running into
issues please file a report at https://github.com/ray-project/ray/issues.
```

And still skip the tests like in the rest of the PR.

Comment (mattip) [reply python/ray/cluster_utils.py]: done

**Comment by mattip** [comment]
@pcmoritz I think this is ready for another review

**Comment by mattip** [comment]
I don't think the failing CI runs are connected with this change. @pcmoritz ping

**Comment by mattip** [comment]
@pcmoritz ping.

File: python/ray/tests/test_ray_init.py [hunk 2]
```diff
@@ -70,6 +70,7 @@ def f():
             host=redis_ip, port=redis_port, password=password)
         assert redis_client.ping()
 
+    @pytest.mark.xfail(cluster_not_supported, reason="cluster not supported")
```
Comment (pcmoritz) [inline python/ray/tests/test_ray_init.py:73]: Let's keep this test, as it is not truly using a cluster :)

File: python/ray/tests/test_advanced.py [hunk 3]
```diff
@@ -375,7 +375,7 @@ def test_multithreading(ray_start_2_cpus):
     # This test requires at least 2 CPUs to finish since the worker does not
     # release resources when joining the threads.
 
-    def run_test_in_multi_threads(test_case, num_threads=10, num_repeats=25):
+    def run_test_in_multi_threads(test_case, num_threads=4, num_repeats=25):
```
Comment (pcmoritz) [inline python/ray/tests/test_advanced.py:378]: Let's revert the changes in this file (and skip it on windows if needed)

**Review by pcmoritz — APPROVED** [review]
This looks good to me now, let's fix the last two comments I made and merge it! Thanks a lot for digging into this :)

**Comment by mattip** [comment]
Fixed the review comments and CI has completed

**Comment by pcmoritz** [comment]
Re-running the tests on latest master, as the previous one failed due to unrelated issues that are already fixed in master.
